lib/plugins/filter/after_render/meta_generator.js

```diff
@@ -1,13 +1,22 @@
 'use strict';
 
+const { Cache } = require('hexo-util');
+const cache = new Cache();
+
 function hexoMetaGeneratorInject(data) {
-  const { config } = this;
-  if (!config.meta_generator
-    || data.match(/<meta([\s]+|[\s]+[^<>]+[\s]+)name=['|"]?generator['|"]?/i)) return;
+  const { config, version } = this;
+
+  const needInject = cache.apply('need-inject', () => {
```

Comment:
I just assume `meta[name="generator"]` will be inserted to every page if `meta_generator()` is used or `<meta name="generator" content="Hexo [version]">` is manually added in the theme.